projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -156,6 +175,34 @@ public function get_publicize_shares_count() {
 		return rest_ensure_response( $response );
 	}
 
+	/**
+	 * Gets information about the currently promoted social product.
+	 *
+	 * @return string|WP_Error A JSON object of the current social product being promoted if the request was successful, or a WP_Error otherwise.
+	 */
+	public static function get_social_promoted_product_info() {
+		$request_url   = 'https://public-api.wordpress.com/rest/v1.1/products?locale=' . get_user_locale() . '&type=jetpack';
+		$wpcom_request = wp_remote_get( esc_url_raw( $request_url ) );
+		$response_code = wp_remote_retrieve_response_code( $wpcom_request );
+		if ( 200 === $response_code ) {
+			$products = json_decode( wp_remote_retrieve_body( $wpcom_request ) );
+			return array(
+				'advanced' => $products->{self::JETPACK_SOCIAL_ADVANCED_YEARLY},
+				'basic'    => $products->{self::JETPACK_SOCIAL_BASIC_YEARLY},
+			);
+		} else {
```

Comment:
This `else` is unnecessary because there's a `return` in the `if` statement already. Personally, I also prefer flipping the logic and returning the `WP_Error` early if the response code isn't `200` but that's more of a personal preference.

What happens if one of the products is missing (e.g. `$products->{self::JETPACK_SOCIAL_ADVANCED_YEARLY}`)? Does the page get a fatal error or does it just give an infinite spinner?

It might be nice to add some kind of fallback for that, but if it just keeps spinning (no error) I think that's okay.

Reply:
> This else is unnecessary

Yeah nice catch, to be honest I grabbed this from Backup. I should have noticed this!

> What happens if one of the products is missing

Right now it's infinite spinner, I am not sure what to do then, we could show a notice or something, but that could be another change with design involved, what do you think? I changed now, that if only one is missing, we show the spinner infinitely only for that one.